habitat_sim/utils/viz_utils.py

```diff
@@ -94,3 +99,161 @@ def display_video(video_file: str, height: int = 400):
+        "dims": overlay dimensions (Tuple : (width, height))\n
+        "pos": overlay position (top left) (Tuple : (width, height))\n
+        "border": overlay image border thickness (int)\n
+        "border_color": overlay image border color [0-255] (3d: array, list, or tuple). Defaults to gray [150]\n
+        "obs": observation key (string)\n
+    """
+    videodims = observations[0][primary_obs].shape
+    videodims = (videodims[1], videodims[0])  # flip to w,h order
+    if not video_file.endswith(".mp4"):
+        video_file = video_file + ".mp4"
+    print("Encoding the video: %s " % video_file)
+    writer = get_fast_video_writer(video_file, fps=fps)
+
+    # build the border frames for the overlays and validate settings
+    border_frames = []
+    if overlay_settings is not None:
+        for overlay in overlay_settings:
+            border_image = np.zeros(
+                (
+                    overlay["dims"][1] + overlay["border"] * 2,
+                    overlay["dims"][0] + overlay["border"] * 2,
+                    3,
+                ),
+                np.uint8,
+            )
+            border_color = np.ones(3) * 150
+            if "border_color" in overlay:
+                border_color = np.asarray(overlay["border_color"])
+            border_image[:, :] = border_color
+            border_frames.append(observation_to_image(border_image, "color"))
+
+    for ob in observations:
+        # primary image processing
+        image_frame = observation_to_image(ob[primary_obs], primary_obs_type)
+        if image_frame is None:
+            print("make_video_new : Aborting, primary image processing failed.")
+            return
+
+        # overlay images from provided settings
+        if overlay_settings is not None:
+            for ov_ix, overlay in enumerate(overlay_settings):
+                overlay_rgb_img = observation_to_image(
+                    ob[overlay["obs"]], overlay["type"], depth_clip
+                )
+                if overlay_rgb_img is None:
+                    print(
+                        'make_video_new : Aborting, overlay image processing failed on "'
+                        + overlay["obs"]
+                        + '".'
+                    )
+                    return
+            overlay_rgb_img = overlay_rgb_img.resize(overlay["dims"])
+            image_frame.paste(
+                border_frames[ov_ix],
+                box=(
+                    overlay["pos"][0] - overlay["border"],
+                    overlay["pos"][1] - overlay["border"],
+                ),
+            )
+            image_frame.paste(overlay_rgb_img, box=overlay["pos"])
+
+        if video_dims is not None:
+            image_frame = image_frame.resize(video_dims)
+
+        # write the desired image to video
+        writer.append_data(np.array(image_frame))
+
+    writer.close()
+    if open_vid:
+        print("Displaying video")
```

Comment:
Still ambivalent of putting a print function in utiliy code... otherwise this func looks good.